### DISCUSSION TITLE
React Library bundled with TSUP causing useState undefined

### DISCUSSION THREAD

**Discussion by rreeves8** [remix-run/remix#9253] [Q&A]
Hi

I'm having an issue with a custom library we built for our remix application. 

I'm bundling custom hooks that use react hooks using tsup. When I bundle them using TSUP and import them into the react application. The custom hooks are unable to use the imported react hooks throwing: TypeError: Cannot read properties of null (reading 'useState')

Does anyone know how to correctly configure tsup to fix the react script bundling error ? Or maybe this is an issue with the remix compiler ? 

Thanks

**Comment by suhaotian** [comment]
Do you want try bunchee, My npm libs already use bunchee, and work well.

These 3 libs use bunchee, maybe you can get some inspire:

https://github.com/suhaotian/remix-intl/tree/main/remix-intl
https://github.com/suhaotian/use-one
https://github.com/suhaotian/xior

Any issues with bunchee, welcome to ask.

**Reply by philrez** [reply]
Good marketing. 👍
